src/eat.c

```diff
@@ -3373,6 +3375,15 @@ vomit(void) /* A good idea from David Neves */
            reaches 0, but only if u.uhs < FAINTING (and !cantvomit()) */
         if (u.uhs >= FAINTING)
             Your("%s heaves convulsively!", body_part(STOMACH));
+         /* Vomiting on an altar is, all things considered, rather impolite. */
+        if (IS_ALTAR(levl[u.ux][u.uy].typ)) {
+            altar_wrath(u.ux, u.uy);
+        }
+        if (acidic(g.youmonst.data)) {
+            if (is_ice(u.ux, u.uy)) {
+                melt_ice(u.ux, u.uy, "Your stomach acid melts straight through the ice!");
+            }
+        }
```

Comment:
This part doesn't make sense to me. Even if it's super powerful acid (which to my knowledge doesn't have incredibly melty effects on ice), you're not really adding enough thermal energy to the ice to turn it back into a deep pool of water.

Reply:
That's a fair point; I was enticed by the idea of ice/vomit-based YASDs. I wonder if there's an alternate way this could be implemented.

Reply:
> 
> 
> This part doesn't make sense to me. Even if it's super powerful acid (which to my knowledge doesn't have incredibly melty effects on ice), you're not really adding enough thermal energy to the ice to turn it back into a deep pool of water.

Interesting. Wouldn't the acid be hot after it left the player's stomach? That thermal energy would melt some of the ice, and the rest would be produced in the resulting reaction. Also, since the acid is so strong and there is a significant quantity of it, it COULD produce an explosion...